app/src/test/java/org/simple/clinic/patient/PatientSyncTest.kt

```diff
@@ -1,121 +0,0 @@
-package org.simple.clinic.patient
-
-import com.f2prateek.rx.preferences2.Preference
-import com.nhaarman.mockito_kotlin.any
-import com.nhaarman.mockito_kotlin.mock
-import com.nhaarman.mockito_kotlin.never
-import com.nhaarman.mockito_kotlin.verify
-import com.nhaarman.mockito_kotlin.whenever
-import io.reactivex.Completable
-import io.reactivex.Observable
-import io.reactivex.Single
-import org.junit.After
-import org.junit.Before
-import org.junit.Test
-import org.mockito.Mockito.reset
-import org.simple.clinic.patient.SyncStatus.DONE
-import org.simple.clinic.patient.SyncStatus.INVALID
-import org.simple.clinic.patient.SyncStatus.IN_FLIGHT
-import org.simple.clinic.patient.SyncStatus.PENDING
-import org.simple.clinic.patient.sync.PatientPullResponse
-import org.simple.clinic.patient.sync.PatientSync
-import org.simple.clinic.patient.sync.PatientSyncApiV1
-import org.simple.clinic.sync.DataPushResponse
-import org.simple.clinic.sync.SyncConfig
-import org.simple.clinic.sync.ValidationErrors
-import org.simple.clinic.util.None
-import org.simple.clinic.util.Optional
-import org.threeten.bp.Instant
-import java.util.Collections
-import java.util.UUID
-
-class PatientSyncTest {
-
-  private val api: PatientSyncApiV1 = mock()
-  private val repository: PatientRepository = mock()
-  private val lastSyncTimestamp: Preference<Optional<Instant>> = mock()
-
-  // This is a lambda just so that the config can be changed later.
-  private val syncConfigProvider: () -> SyncConfig = mock()
-
-  private lateinit var patientSync: PatientSync
-
-  @Before
-  fun setup() {
-    patientSync = PatientSync(api, repository, Single.fromCallable { syncConfigProvider() }, lastSyncTimestamp)
-  }
-
-  @Test
-  fun `when pending-sync patients are empty then the server should not get called`() {
-    whenever(repository.patientsWithSyncStatus(PENDING)).thenReturn(Single.just(Collections.emptyList()))
-
-    patientSync.push().blockingAwait()
-
-    verify(api, never()).push(any())
-  }
-
-  @Test
-  fun `when a push succeeds then any patients that were created or updated during the push started should not get marked as synced`() {
-    // TODO: I'm out of ideas to write this test because it involves concurrency.
-  }
-
-  @Test
-  fun `errors during push should not affect pull`() {
```

Comment:
This (and the next function) seemed unnecessary to test since we are only testing whether `Completable.mergeArrayDelayError` works.

Reply:
No, my intention was to test that errors in the implementation get swallowed even if someone replaces `mergeArrayDelayError` with something else.

Reply:
Got it. The tests which were left out have now been implemented by the `ModelSyncTest` class.